### ISSUE TITLE
Schema delegation to contained dict is incomplete

### ISSUE THREAD

**Issue by RossBoylan** [JuliaStats/StatsModels.jl#193]
The comment in `schema.jl` says "A `Schema` behaves for all intents and purposes like an immutable `Dict`" and the user documentation says something similar.  But it's not really true.  The comment goes on to note 

> , and delegates the constructor as well as `getindex`, `get`, `merge!`, `merge`,
> `keys`, and `haskey` to the wrapped `Dict`.

which translates to "behaves like a `Dict` if you call those methods and otherwise not."

I noticed this because about a year ago, with v 0.6.5 I was able to use `values` to pull out members of the schema, but now this produces an undefined function error.  https://discourse.julialang.org/t/statsmodels-values-schema-no-longer-works/46659 provides more background.  As noted there it is not clear to me why it stopped working, because even the old version did not define `values`.

My particular problem can be solved by defining
```julia
Base.values(schema::Schema) = values(schema.schema)
```
but that doesn't solve the general problem that `Schema` does not act like a `Dict` in many ways (e.g., my sample code on discourse also shows a failure because `iterate` is not defined, triggered by the use of `foreach`).

I don't understand the design goals well enough to know whether implementing the full `Dict` protocol is a good idea, but the current implementation seems to be at an uncomfortable part-way there point.

**Comment by kleinschmidt** [comment]
Yeah it's spotty.  At one point the schema really *was* just a `Dict` but that lead to too many method ambiguity errors so now there's a special type for it.

I'd be open to a PR which implements the missing methods, they can just be passed through like you noted.  I just have never needed it internally but if it's useful to you than I don't object!

**Comment by RossBoylan** [comment]
Would it be better to implement the `Dict` protocol (I'm not entirely sure how to discover what that is; it doesn't seem to be explicitly documented in https://docs.julialang.org/en/v1/base/collections/#Base.Dict), e.g., `keys` and `values`, or use domain-specific names like `abstractTerms` and `concreteTerms`?

Do you have any ideas why `values` was working before?

**Comment by kleinschmidt** [comment]
See #103: before that, there was no `Schema`, just a `Dict`.

I think it's better to implement the "normal" methods for `Dict` rather than creating our own special ones...
